core/src/test/java/com/datastax/oss/driver/internal/core/CompletionStageAssert.java

```diff
@@ -66,8 +66,31 @@ public CompletionStageAssert(CompletionStage<V> actual) {
     return isFailed(f -> {});
   }
 
+  public CompletionStageAssert<V> isCancelled() {
+    assertThat(actual.toCompletableFuture().isCancelled())
+        .overridingErrorMessage("Expected completion stage to be cancelled")
+        .isTrue();
```

Comment:
It might be good to have a small delay in here, I've had similar assertions fail in CI because of timing issue (see 82d0d280078cb10ec043c544c9cd7e3d8dea6b02).